transport/src/main/java/io/camunda/zeebe/transport/stream/impl/RemoteStreamRegistry.java

```diff
@@ -82,11 +95,17 @@ public void remove(final UUID streamId, final MemberId receiver) {
     final var uniqueId = new StreamId(streamId, receiver);
     final var consumer = idToConsumer.remove(uniqueId);
     if (consumer != null) {
-      typeToConsumers.computeIfPresent(
-          consumer.streamType(),
-          (type, consumerSet) -> {
-            consumerSet.remove(consumer);
-            return consumerSet.isEmpty() ? null : consumerSet;
+      final var logicalId = new LogicalId<>(consumer.streamType(), consumer.properties());
```

Comment:
:upside_down_face: I think on the client side we kept the logical ID on the consumer as well to avoid recreating it here, but this is super minor optimization (and for all I know, it might be entirely escaped at runtime :shrug:)